app/models/following.rb

```diff
@@ -6,4 +6,14 @@ def initialize(following_data)
     @handle = following_data[:login]
     @url = following_data[:html_url]
   end
+
+  def in_system?
+    @friend = User.find_by(handle: @handle)
+    @friend.present?
+  end
+
+  def already_friends?(user_id)
```

Comment:
I take it this prevents friends from being added twice

Reply:
that probably would be useful also, but this is only checking whether a friendship exists, and if not to display the button. I don't think that a duplicate friendship would be able to be created as a result of the button being taken away but then again I know nothing about back doors.